tests/pyfunc/test_spark.py

```diff
@@ -77,7 +77,8 @@ def model_path(tmpdir):
 
 
 @pytest.mark.large
-def test_spark_udf(spark, model_path):
+@pytest.mark.parametrize("env_manager", ["local", "conda"])
```

Comment:
Why do we need to parametrize every single test with `env_manager`?

Reply:
I add a new test for restoring conda env case, and keep old tests unchanged.